app/src/main/java/com/tz/sanga/moviestore/Fragments/FirstFragment.java

```diff
@@ -0,0 +1,254 @@
+                        progressBar.setVisibility(View.GONE);
+                        return false;
+                    }
+
+                    @Override
+                    public boolean onResourceReady(GlideDrawable resource, String model, Target<GlideDrawable> target, boolean isFromMemoryCache, boolean isFirstResource) {
+                        progressBar.setVisibility(View.GONE);
+                        return false;
+                    }
+                })
+                .centerCrop()
+                .into(imageView);
+    }
+
+    public void addToSqlDB(final View view) {
+        favoriteDb = new FavoriteDb(getContext());
+        if (!favoriteDb.checkMovie(path)){
+            boolean insertData = favoriteDb.addFavorite(moveId, title, overview, path);
+            if (insertData){
+                Snackbar.make(view, title + " added to favorite", Snackbar.LENGTH_LONG).setAction("REMOVE", new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {deleteFavorite(v);}
+                }).show();
+            }else {Snackbar.make(view, title + " not added to favorite", Snackbar.LENGTH_LONG).show();}
+        }
+        else {Snackbar.make(view, title + " exist to favorite", Snackbar.LENGTH_LONG).setAction("REMOVE", new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {deleteFavorite(v);}
+            }).show();
+        }
+    }
+
+    public void deleteFavorite(View view){
+        boolean delete = favoriteDb.deleteMovie(path);
+        if (delete){Snackbar.make(view, "Deleted successful", Snackbar.LENGTH_LONG).show();}
+        if (!delete){ Snackbar.make(view, "Not successful", Snackbar.LENGTH_LONG).show();}
+    }
+
+    @Override
+    public void showLoading() {progressBar.setVisibility(View.VISIBLE);}
+
+    @Override
+    public void hideLoading() {
+        progressBar.setVisibility(View.INVISIBLE);
+    }
+
+    @Override
+    public void showResults(List<Movie> moveData) {
+        if (moveData.size()<1){
+            textView1.setVisibility(View.GONE);
+            multiSnapRecyclerView.setVisibility(View.GONE);
+        }
+        textView1.setText(R.string.related_movies);
+        adapter.addAll(moveData);
+    }
+
+    @Override
+    public void onReload(int moveId, String path, String overview, String title, String date) {
+        setData(moveId,path,overview,title, date);
+        scrollView.fullScroll(View.FOCUS_BACKWARD);
+        presenter.updateMoveId(moveId);
+        presenter.getData();
+        setView();
+        presenter.processTrailer();
+    }
+
+    private void trailer(){
+        List<Trailer> moveData;
+        TrailerAdapter trailerAdapter;
+        moveData = new ArrayList<>();
+        layoutManager = new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false);
+        trailerAdapter = new TrailerAdapter(getContext(), moveData);
```

Comment:
Don't initialize adapter more than once, do it once and used notify data set change to update the specific adapter details